src/conversion/model-to-view-converters.js

```diff
@@ -288,6 +289,157 @@ export function unwrap( elementCreator ) {
 }
 
 /**
+ * Function factory, creates a converter that converts new model marker to view attribute element.
+ *
+ * Let's assume following model and view. `{}` represents a range that is added as a marker with `searchResult` name.
+ * The range represents a result of search `ab` string in the model document. The range has to be visualized in view.
+ *
+ *		[paragraph]              MODEL ====> VIEW        <p>
+ *			|- {a                                         |- <span class="searchResult">
+ *			|-  b}                                        |   |- ab
+ *			|-  c                                         |- c
+ *
+ * The wrapping node depends on passed parameter. If {@link engine.view.AttributeElement} was passed, it will be cloned and
+ * the copy will become the wrapping element. If `Function` is provided, it is passed all the parameters of the
+ * {@link engine.conversion.ModelConversionDispatcher#event:addMarker addMarker event}. It's expected that the
+ * function returns a {@link engine.view.AttributeElement}. The result of the function will be the wrapping element.
+ * When provided `Function` does not return element, then will be no conversion.
+ *
+ * The converter automatically consumes corresponding value from consumables list, stops the event (see
+ * {@link engine.conversion.ModelConversionDispatcher}).
+ *
+ *		modelDispatcher.on( 'addMarker:searchResult', wrapMarker( new ViewAttributeElement( 'span', { class: 'searchResult' } ) ) );
+ *
+ * @external engine.conversion.modelToView
+ * @function engine.conversion.modelToView.wrapMarker
+ * @param {engine.view.AttributeElement|Function} elementCreator View attribute element, or function returning
+ * a view attribute element, which will be used for wrapping.
+ * @returns {Function} Add marker converter.
+ */
+export function wrapMarker( elementCreator ) {
```

Comment:
As far as name is concerned, I agree that the name could be different, i.e. `wrapRange` (while `wrap` could be changed to `wrapItem`, same for `unwrap`s).

The functions (`wrap` and `wrapMarker`) could be merged. But there are couple of differences between them:
1. Consumable type. It's taken from `eventNameToConsumableType( evt.name )` and would be invalid for `wrapMarker`. This would have to change.
2. `data` object passed to the converter is different. We can try to unify them a bit though. `wrap` gets: `item`, `range`, `attributeKey`, `attributeOldValue` and `attributeNewValue`. `wrapMarker` gets `range` and `name`. We could add `data.item` to `wrapMarker` `data` and make it equal to `data.range`, but that's it.
3. There's difference if view element is created using "element creator function":
`wrap`: `elementCreator( data.attributeNewValue, data, consumable, conversionApi );`
`wrapMarker`: `elementCreator( data, consumable, conversionApi );`
It would have to be something like:
`elementCreator( data.attributeNewValue || data.name, data, consumable, conversionApi );` and still, `elementCreator` functions would have to be different for converting items and ranges which means that we need to write about it in docs and it might be confusing.
4. Both converters have special behavior that are individual for them. `wrap` has to "unwrap" part of view if attribute changed. `wrapMarker` has to support collapsed ranges. Those would be just two `if`s standing next to each other, operating on different values and "active" depending on `data` content. We can put them together in one function, but on the same basis we could put everything into one class.

All in all the merging would not be as seamless as it looks on first sight so I am against it. Let's change `wrapMarker` to `wrapRange` and `wrap` to `wrapItem`, same for `unwrap` and keep them separated.